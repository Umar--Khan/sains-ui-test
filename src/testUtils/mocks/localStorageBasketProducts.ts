import { BasketProduct } from 'src/views/Store/Store.d';

export const mockedLocalStorageBasketProducts = [
  {
    productDetail: {
      'productId': '5493179',
      'sku': '549/3179',
      'title': 'Russell Hobbs Pennine Illuminating S Steel Kettle 20444',
      'price': 24.99,
      'image': 'https://media.4rgos.it/s/Argos/9805244_R_SET?w=110&h=130',
    },
    quantity: 5,
  },
  {
    productDetail: {
      'productId': '4251824',
      'sku': '425/1824',
      'title': 'Challenge White Desk Fan - 12 Inch',
      'price': 14.99,
      'image': 'https://media.4rgos.it/s/Argos/4251824_R_SET?w=110&h=130',
    },
    quantity: 1,
  },
  {
    productDetail: {
      'productId': 'tuc132404445',
      'sku': 'tuc132404445',
      'title': 'Plain Vest Top 12 BLACK',
      'price': 16,
      'image': 'https://media.4rgos.it/s/Argos/tuc132404445_R_SET?w=110&h=130',
    },
    quantity: 2,
  },
] as BasketProduct[];
