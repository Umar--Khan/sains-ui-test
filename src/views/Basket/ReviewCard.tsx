import Card from 'src/shared/Card/Card';
import Button from 'src/shared/Buttons/Button/Button';
import { BasketProduct } from '../Store/Store.d';
import { ChangeEvent } from 'react';
import Select from 'src/shared/Fields/Select/Select';

export interface ReviewCardProps {
  basketProduct: BasketProduct;
  handleOnClick: (arg: BasketProduct) => void;
  handleQtySelect: (
    event: ChangeEvent<HTMLSelectElement>,
    productId: string,
  ) => void;
}

const ReviewCard = ({
  basketProduct,
  handleOnClick,
  handleQtySelect,
  ...rest
}: ReviewCardProps) => {
  const {
    productDetail: { image, title, price, productId },
    quantity,
  } = basketProduct;

  const renderQtyOptions = () => {
    const optionsArr = Array.from(Array(quantity + 5).keys());

    return optionsArr.map(
      quantity =>
        quantity >= 1 && (
          <option value={quantity} key={quantity}>
            {quantity}
          </option>
        ),
    );
  };
  return (
    <Card key={productId} {...rest}>
      <img src={image} alt={title} />
      <p>
        {title}, &#163;{price}
      </p>
      <p>
        Qty:{' '}
        <Select
          onChange={e => handleQtySelect(e, productId)}
          value={quantity}
          options={renderQtyOptions()}
          name='qty'
        />
      </p>

      <p>Total price: &#163;{(quantity * price).toFixed(2)}</p>
      <Button onClick={() => handleOnClick(basketProduct)}>
        Remove from basket
      </Button>
    </Card>
  );
};

export default ReviewCard;
