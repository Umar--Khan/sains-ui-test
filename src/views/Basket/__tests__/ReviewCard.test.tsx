import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import ReviewCard, { ReviewCardProps } from '../ReviewCard';
import { mockedLocalStorageBasketProducts } from 'src/testUtils/mocks/localStorageBasketProducts';

describe('<ReviewCard />', () => {
  const props = {
    basketProduct: mockedLocalStorageBasketProducts[0],
    handleOnClick: jest.fn(),
  } as ReviewCardProps;

  it('Should render correctly', () => {
    const component = renderer.create(<ReviewCard {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should call the onClick prop', () => {
    render(<ReviewCard {...props} />);
    fireEvent.click(screen.getByText('Remove from basket'));
    expect(props.handleOnClick).toHaveBeenCalledTimes(1);
    expect(props.handleOnClick).toHaveBeenCalledWith(
      mockedLocalStorageBasketProducts[0],
    );
  });
});
