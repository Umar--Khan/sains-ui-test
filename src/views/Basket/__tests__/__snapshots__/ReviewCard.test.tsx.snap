// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCard /> Should render correctly 1`] = `
<div
  className="sc-bdfBwQ fvoACg"
>
  <img
    alt="Russell Hobbs Pennine Illuminating S Steel Kettle 20444"
    src="https://media.4rgos.it/s/Argos/9805244_R_SET?w=110&h=130"
  />
  <p>
    Russell Hobbs Pennine Illuminating S Steel Kettle 20444
    , £
    24.99
  </p>
  <p>
    Qty:
     
    <select
      name="qty"
      onChange={[Function]}
      value={5}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
      <option
        value={6}
      >
        6
      </option>
      <option
        value={7}
      >
        7
      </option>
      <option
        value={8}
      >
        8
      </option>
      <option
        value={9}
      >
        9
      </option>
    </select>
  </p>
  <p>
    Total price: £
    124.95
  </p>
  <button
    className="sc-gsTCUz kiZIlZ"
    onClick={[Function]}
  >
    Remove from basket
  </button>
</div>
`;
